Assets/Scripts/Interactables/Resource/Resource.cs

```diff
@@ -0,0 +1,89 @@
+using Examen.Inventory;
+using Examen.Items;
+using Examen.Poolsystem;
+using FishNet.Object;
+using MarkUlrich.Health;
+using UnityEngine;
+
+namespace Examen.Interactables.Resource
+{
+    [RequireComponent(typeof(HealthData))]
+    public class Resource : NetworkBehaviour, Interactable
+    {
+        [HideInInspector] public HealthData HealthData;
+        [HideInInspector] public PoolSystem poolSystem;
+        [HideInInspector] public bool HasHealthData;
+
+        public Item ResourceItem;
+        public int AmountToGive = 1;
+        public int DeathTime;
+
+        public const int DamageAmount = 1;
+        /// <summary>
+        /// Sets PoolSystem and HealthData Variables and add StartDeathTimer to the onDie Event.
+        /// </summary>
+        [Server]
+        public virtual void Start()
+        {
+            poolSystem = PoolSystem.Instance;
+            HealthData = GetComponent<HealthData>();
+            HealthData.onDie.AddListener(StartDeathTimer);
+            HasHealthData = true;
+        }
+
+        /// <summary>
+        /// Resurrect the player if it has healthData.
+        /// </summary>
+        [Server]
+        public virtual void OnEnable()
+        {
+            if(HasHealthData)
+                HealthData.Resurrect(HealthData.MaxHealth);
+        }
+
+        /// <summary>
+        /// Calls on every functionality, that needs to happen when interacting with the resources. 
+        /// </summary>
+        [ServerRpc]
+        public virtual void Interact()
+        {
+            if (!IsOwner)
+                return;
```

Comment:
Is owner check cannot be done in a method with the serverRPC tag, serverRPC is basically only for sending data to a server method